common/api/appui-react.api.md

```diff
@@ -3895,6 +3944,20 @@ export class SheetsModalFrontstage implements ModalFrontstageInfo {
     title: string;
 }
 
+// @public (undocumented)
+export interface ShowComponentOptions {
+    // (undocumented)
+    anchorRef?: React_2.RefObject<HTMLElement>;
+    // (undocumented)
+    location: XAndY;
+    // (undocumented)
+    offset: XAndY;
+    // (undocumented)
+    onCancel: OnCancelFunc;
+    // (undocumented)
+    relativePosition: RelativePosition;
```

Comment:
Taking a look now. Inferring that:
![image](https://github.com/iTwin/appui/assets/78756012/55fd9bd4-2589-4450-9fc6-de8d3fb3f20d)
`RightTop` and `LeftTop` are "actually" alignments? Meaning your "start" and  "end" literals?

Reply:
First attempt pushed. It's not quite what you had in mind, so let me know if you're set on the `dashed-syntax` and why exactly you were interested in the string template literals. 

I'm also confused by RelativePosition in general - TopLeft vs LeftTop for instance. I see how the difference is calculated between them (in [Popup.tsx](https://github.com/iTwin/appui/blob/6bcc75cec129d5c08f86edff91ce736c34cf8796/ui/core-react/src/core-react/popup/Popup.tsx#L376) for example) but I don't understand semantically what is going on. Why, for instance, is there no `LeftBottom` or `RightBottom`?  Why is the `offset` only considered sometimes? 

Any clarification would be helpful :)

Reply:
I wasn't advocating to try and match the existing `RelativePosition` types to new literal values for new APIs (since I agree that they do not make much sense semantically..).
Seems like side (left/right/top/bottom) + placement (start/end) should be enough i.e. https://github.com/floating-ui/floating-ui/blob/495954da10d2abe8a5e6a0f0247582f071d6f031/packages/utils/src/index.ts#L4